grift/loaders.py

```diff
@@ -67,6 +69,96 @@ def reload(self):
         self._source = os.environ
 
 
+class VaultException(Exception):
+    """Custom exception raised if Vault response contains error message"""
+
+
+class VaultTokenLoader(DictLoader):
+    def __init__(self, url, path, token):
+        """Load secrets from a Vault path, using token authentication
+
+        See https://www.vaultproject.io/docs
+
+        Args:
+            url: Vault url
+            path: Vault path to fetch secrets from
+            vault_token: token (must have access to vault path)
+        """
+        self._vault_url = url
+        self._path = path
+        self._token = token
+        source_dict = self._fetch_secrets()
+        super(VaultTokenLoader, self).__init__(source_dict)
+
+    @property
+    def _headers(self):
+        """Return token header to access vault"""
+        return {'X-Vault-Token': self._token}
+
+    def _fetch_secrets(self):
+        """Read data from the vault path"""
+        url = '{}/v1/{}'.format(self._vault_url, self._path)
```

Comment:
Thanks for catching this. I ended up avoided urljoin because it's kind of awkward with 2/3, and because it's not super helpful in joining multiple segments